ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/TaskList/Decision/DecisionDate.cshtml.cs

```diff
@@ -25,21 +26,25 @@ public class DecisionDate : DecisionBaseModel
 		[Required(ErrorMessage = "Please enter a decision date"), Display(Name = "Decision")]
 		public DateTime? DateOfDecision { get; set; }
 
+		  public string DecisionText { get; set; }
+
 		public LinkItem GetPageForBackLink(int id)
 		{
 			var decision = GetDecisionFromSession(id);
 
-			return decision.ApprovedConditionsSet switch
+			return decision.Decision switch
 			{
-				true => Links.Decision.WhatConditions,
-				_ => Links.Decision.AnyConditions,
+				AdvisoryBoardDecisions.Approved => decision is { ApprovedConditionsSet : true } ? Links.Decision.WhatConditions : Links.Decision.AnyConditions,
+				AdvisoryBoardDecisions.Declined => Links.Decision.DeclineReason,
+				_ => throw new Exception("Unexpected decision state")
 			};
 		}
 
 		public async Task<IActionResult> OnGetAsync(int id)
 		{
 			await SetDefaults(id);
 			DateOfDecision = GetDecisionFromSession(id)?.AdvisoryBoardDecisionDate;
+			DecisionText = GetDecisionFromSession(id)?.Decision.ToString().ToLowerInvariant();
```

Comment:
I think you need a couple more question marks here - 
`DecisionText = GetDecisionFromSession(id)?.Decision?.ToString()?.ToLowerInvariant();`

Reply:
There's going to be a 'redirect to the start of the journey' guard clause before this - as discussed in Slack this morning, so shouldn't be a problem - will always be populated by the time this executes.